src/shared/mkfs-util.c

```diff
@@ -63,24 +65,45 @@ int make_filesystem(
                         return log_oom();
         }
 
+        if (streq(fstype, "vfat")) {
+                /* Classic FAT only allows 11 character uppercase labels */
+                strncpy(mangled_label, label, sizeof(mangled_label)-1);
+                mangled_label[sizeof(mangled_label)-1] = 0;
+                ascii_strupper(mangled_label);
+                label = mangled_label;
+
+                xsprintf(vol_id, "%08" PRIx32,
+                         ((uint32_t) uuid.bytes[0] << 24) |
+                         ((uint32_t) uuid.bytes[1] << 16) |
+                         ((uint32_t) uuid.bytes[2] << 8) |
+                         ((uint32_t) uuid.bytes[3])); /* Take first 32 bytes of UUID */
+        }
+
+        if (isempty(vol_id))
```

Comment:
maybe just replace this `if` line with a simple `else`?

Reply:
In the next commit the `if` above is changed to `if ... else`, so `else` wouldn't work here.